### ISSUE TITLE
add CASE

### ISSUE THREAD

**Issue by CptMoore** [CptMoore/MechEngineer#89]
case:
- (torso) explosion doesn't produce injury
- ammo explosions dont detroy other compartments

**Comment by LadyAlekto** [comment]
That would be CASE II

Currently mechs all have CASE

Youd have to start by making a ammo explosion deal all the remaining shots it has stored as damage 

Then add CASE

Then CASE II 

;)

**Comment by GentlemanReaper** [comment]
But do we really need ammo explosions to be that lethal? We could have them as they are, have CASE change it so you receive 33% of the location's max structure in damage, and have CASE II almost nullify it completely.

**Comment by LadyAlekto** [comment]
If id have a choice, id definitely make players pay to have case

**Comment by CptMoore** [comment]
I would implement this and put it on the list of features to extract out of ME. but I dont understand CASE or CASE II or what Lady tells us here.

Ammo explosions currently go to adjacent compartments, so no CASE I or II?!

**Comment by LadyAlekto** [comment]
Currently ammo explosions are coded as a simple true/false event that does "nuke this location"

In TT an ammo explosion follows the damage rules and would go to each adjacent location towards ct

CASE then allowed for ammo to only destroy the location the ammo was stored in, had a weight of half ton and 1 slot per location

CASEII went further, had more weight and safely vented the ammo explosion without any destruction (fis tier3 tech)

**Comment by CptMoore** [comment]
I think I only need 2 patches, one to fix ammoboxes DamageComponent to use DamageLocation (which does the damage transfer), and then in DamageLocation itself a way to ignore armor in case of ammo explosions.

both have to use transpiler though.
maybe need 2 additional patches to retain context information about if its an ammo explosion or not.

**Comment by CptMoore** [comment]
```
BattleMechs can survive the destruction of any body section
except the head or center torso. If a section is destroyed
and the same location takes another hit, or if excess damage
remains from the shot that destroyed the location, that damage
transfers to (affects) the outer armor of the next location inward.
Excess ammunition explosion damage is transferred directly to
the internal structure of the next location inward
```

**Comment by CptMoore** [comment]
implemented this with e0a1e14 

issue are when the floaties come up, have to put the blowback floatie earlier
CASE still costs nothing are weights nothing

**Comment by CptMoore** [comment]
addressed previous issues in a7baf12
